NitroxPatcher/Patches/Dynamic/Fire_Douse_Patch.cs

```diff
@@ -1,30 +1,73 @@
-﻿using System.Reflection;
+using System.Reflection;
 using HarmonyLib;
+using NitroxClient.Communication.Abstract;
 using NitroxClient.GameLogic;
-using NitroxModel.Core;
+using NitroxClient.Helpers;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
 using NitroxModel.Helper;
+using NitroxModel.Packets;
 
-namespace NitroxPatcher.Patches.Dynamic
+namespace NitroxPatcher.Patches.Dynamic;
+
+public class Fire_Douse_Patch : NitroxPatch, IDynamicPatch
 {
-    public class Fire_Douse_Patch : NitroxPatch, IDynamicPatch
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((Fire t) => t.Douse(default(float)));
+
+    public static void Postfix(Fire __instance)
     {
-        public static readonly MethodInfo TARGET_METHOD = Reflect.Method((Fire t) => t.Douse(default(float)));
+        NitroxId nitroxId = GetId(__instance);
 
-        public static void Postfix(Fire __instance, float amount)
+        if (nitroxId == null)
         {
-            if (!__instance.livemixin.IsAlive() || __instance.IsExtinguished())
-            {
-                NitroxServiceLocator.LocateService<Fires>().OnDouse(__instance, 10000);
-            }
-            else
-            {
-                NitroxServiceLocator.LocateService<Fires>().OnDouse(__instance, amount);
-            }
+#if DEBUG
+            Log.Error($"Fire instance is unknown to the server {__instance.gameObject.name}");
+#endif
+            return;
         }
 
-        public override void Patch(Harmony harmony)
+        if (!__instance.livemixin.IsAlive() || __instance.IsExtinguished())
+        {
+            // Ensure one final metadata packet goes out to trigger any residual extinguish event that need to happen.
+            // We want to make sure this packet happens before the delete, so send it directly and not with throttling.
+            Resolve<ThrottledPacketSender>().RemovePendingPackets(nitroxId);
+            Resolve<Entities>().EntityMetadataChanged(__instance, nitroxId);
+
+            // Then tell the serve to delete the entity as it is now extinguished. 
+            Resolve<IPacketSender>().Send(new EntityDestroyed(nitroxId));
```

Comment:
Is there a chance for race conditions and could it cause problems? Maybe we want to wait a little before deleting the entity?